### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by rizar**
While working on an attention mechanism demo I came across a few bugs and had to do a few impovements. An incomplete lists of changes:

* adds some logging

* fixes bug in gradient norm computation

* changes the naming scheme for auxiliary variables #163 

* changes the order of the `IsotropicGaussian` parameters #123 

* partial enables monitoring of not-scalar expression by using the `TakeLast` aggregation scheme

### CODE REVIEW

File: blocks/algorithms/__init__.py [hunk 1]
```diff
@@ -164,17 +167,20 @@ class GradientDescent(DifferentiableCostMinimizer):
     """
     def __init__(self, step_rule=None, gradients=None, **kwargs):
         super(GradientDescent, self).__init__(**kwargs)
-        self.gradients = (
-            gradients if gradients
-            else dict(
-                zip(self.params, tensor.grad(self.cost, self.params))))
+        self.gradients = gradients
+        if not self.gradients:
+            logger.info("Taking the cost gradient")
```
Comment (bartvm) [inline blocks/algorithms/__init__.py]: I would make this `logger.debug` and not `info`, it's pretty detailed information.

Comment (rizar) [inline blocks/algorithms/__init__.py]: Actually taking the gradient can take more then half-a-minute when you are working with scan. So you want this logging to be always on, and you do no want to see those really detailed messages you get at the debug level.

Comment (bartvm) [inline blocks/algorithms/__init__.py]: Fair point

**Comment by bartvm** [comment]
LGTM

**Pull request merged**